src/Query/Collection.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Softonic\GraphQL\Query;
+
+use Softonic\GraphQL\Traits\GqlIterator;
+use Softonic\GraphQL\Traits\JsonSerializer;
+
+class Collection implements ReadObject, \JsonSerializable, \Iterator
+{
+    use GqlIterator, JsonSerializer;
```

Comment:
It's weird to have `GqlIterator` trait implementing an interface

Reply:
It's common code between Query/Item, maybe ReadObject could be an abstract class instead of an interface, and could include the GqlIterator methods.